app/assets/javascripts/discourse/tests/integration/components/prosemirror-editor/link-test.gjs

```diff
@@ -28,6 +28,11 @@ module(
         '<p><a href="https://example.com/file.pdf" class="attachment">File</a></p>',
         "[File|attachment](https://example.com/file.pdf)",
       ],
+      "attachment link with hash upload": [
+        "[File|attachment](upload://some-hash)",
+        '<p><a href="/404" class="attachment" data-orig-href="upload://some-hash">File</a></p>',
```

Comment:
@megothss @davidtaylorhq I've added support to the `upload://hash` format so it still serializes and parses correctly, but I skipped the hash resolution for now, this `/404` is bogus at the moment because we don't have any way within the editor to actually follow this link.